lib/modules/manager/puppet/common.ts

```diff
@@ -0,0 +1,50 @@
+import { regEx } from '../../../util/regex';
+import { parseUrl } from '../../../util/url';
+import type { PackageDependency } from '../types';
+
+export const RE_REPOSITORY_GENERIC_GIT_SSH_FORMAT =
+  regEx(/^(?:git@[^:]*):(.+)$/);
+
+export function parseGitOwnerRepo(
+  git: string,
+  githubUrl: boolean
+): string | PackageDependency {
+  const genericGitSsh = RE_REPOSITORY_GENERIC_GIT_SSH_FORMAT.exec(git);
+
+  if (genericGitSsh) {
+    return genericGitSsh[1].replace(regEx(/\.git$/), '');
+  } else {
+    if (githubUrl) {
+      return git
+        .replace(regEx(/^github:/), '')
+        .replace(regEx(/^git\+/), '')
+        .replace(regEx(/^https:\/\/github\.com\//), '')
+        .replace(regEx(/\.git$/), '');
+    }
+    try {
+      const url = parseUrl(git);
+
+      if (!url) {
+        return invalidUrl(git);
+      }
+
+      return url.pathname
+        .replace(regEx(/\.git$/), '')
+        .replace(regEx(/^\//), '');
+    } catch (err) {
+      return invalidUrl(git);
+    }
+  }
+}
+
+function invalidUrl(git: string): PackageDependency {
+  return {
+    gitRef: true,
+    sourceUrl: git,
+    skipReason: 'invalid-url',
+  };
+}
+
+export function isGithubUrl(git: string, parsedUrl: URL | undefined): boolean {
+  return parsedUrl?.host === 'github.com' || git.startsWith('git@github.com');
```

Comment:
```suggestion
  return parsedUrl?.host === 'github.com' || url.startsWith('git@github.com');
```
is first condition is false, isn't second always false too? as host is different?

Reply:
same like in https://github.com/renovatebot/renovate/pull/14789#discussion_r862052444
`git@github.com:puppetlabs/puppetlabs-stdlib.git` is not a valid URL so parsedUrl is null. In that case the fallback is the git-url that can fulfill the second condition